core/server/api/v2/utils/serializers/input/posts.js

```diff
@@ -79,6 +80,8 @@ module.exports = {
                 });
             }
         }
+
+        frame.data.posts[0] = url.forPost(Object.assign({}, frame.data.posts[0]), frame.options);
```

Comment:
This reads a little strange for me, without knowing what `url.forPost` does it looks like you're setting the post object to a URL 🤔

I think I'd prefer more explicitness inside the serialiser itself so you can look at the serialiser and immediately see which properties are being modified. It would also keep model-specific knowledge outside of a util function (util functions should be small, simple, and single-purpose) and inside the one place which should have model-specific serialisation knowledge.

Reply:
Hmm, I hadn't seen that go by but that also looks odd compared how most serializer layers I've worked with are built. The `withRelated` part inside the `urlFor()` method seems like a bit of a code smell, I would expect embedded records to be passed through the respective serialiser rather than a util function having intimate knowledge of how a model hierarchy might be set up.

Anyhow, seeing as that's already in the codebase maybe it's something to improve in a later cleanup so that we're not blocking.

Reply:
@kevinansfield this `urlFor` functionality is this way based on discussions we had with @kirrg001. Everywhere else in the frame, objects are being passed by reference and modified on each stage. I agree that it would be nicer to just return a set of url's in this case and not modify the passed object, but was following the convention rather than having it a different way just in this place :smiley: 
As for `withRelated` part, that was also a smelly part we decided to deal with separately. It's only for `posts` that we need&support nested object decoration and there's no concept of cross-calling other serializers in the frame atm.

Reply:
:+1: definitely into making this part of code better. We could also touch upon the mutability topic while at it as well.
How would you suggest naming this method/module instead?

Reply:
Like the above suggestion, do we update the naming for input serializers in this PR or as a separate PR where both input/output pattern is updated?

Reply:
Separate PR, let's keep this one focused 🙂